File: ABC_Chapter7_Exercise3/app/src/main/java/com/example/cyu/abc_chapter7_exercise3/MainActivity.java
package com.example.cyu.abc_chapter7_exercise3;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.widget.GridView;
import android.widget.ImageView;
import android.widget.BaseAdapter;
import android.view.View;
import android.view.ViewGroup;
import android.widget.AdapterView;
import android.widget.Toast;
import android.content.Context;

public class MainActivity extends AppCompatActivity {

    Integer[] Dog = {
            R.drawable.dog1,
            R.drawable.dog2,
            R.drawable.dog3,
            R.drawable.dog4,
            R.drawable.dog5,
            R.drawable.dog6
    };

    ImageView pic;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        GridView grid = findViewById(R.id.gridView);
        final ImageView pic = findViewById(R.id.imgLarge);
        grid.setAdapter(new ImageAdapter(this));
        grid.setOnItemClickListener(new AdapterView.OnItemClickListener() {
            @Override
            public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
                Toast.makeText(getBaseContext(), "Selected Species " + (position + 1),
                        Toast.LENGTH_SHORT).show();
                pic.setImageResource(Dog[position]);
            }
        });

    }

    public class ImageAdapter extends BaseAdapter {

        private Context context;

        public ImageAdapter(Context c) {
            context = c;
        }

        @Override
        public int getCount() {
            return Dog.length;
        }

        @Override
        public Object getItem(int position) {
            return null;
        }

        @Override
        public long getItemId(int position) {
            return 0;
        }

        @Override
        public View getView(int position, View convertView, ViewGroup parent) {
            pic = new ImageView(context);
            pic.setImageResource(Dog[position]);
            pic.setScaleType(ImageView.ScaleType.FIT_XY);
            pic.setLayoutParams(new GridView.LayoutParams(330, 330));
            return pic;
        }
    }
}

File: ABC_Chapter7_Example/app/src/main/java/com/example/cyu/abc_chapter7_example/MainActivity.kt
package com.example.cyu.abc_chapter7_example

import android.content.Context
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.view.View
import android.view.ViewGroup
import android.widget.AdapterView
import android.widget.BaseAdapter
import android.widget.GridView
import android.widget.ImageView
import android.widget.Toast

class MainActivity : AppCompatActivity() {

    internal var animals = arrayOf(R.drawable.eagle, R.drawable.elephant, R.drawable.gorilla, R.drawable.panda, R.drawable.panther, R.drawable.polar)

    internal lateinit var pic: ImageView

    override fun onCreate(savedInstanaceState: Bundle?) {
        super.onCreate(savedInstanaceState)
        setContentView(R.layout.activity_main)
        val grid = findViewById<GridView>(R.id.gridView)
        val pic = findViewById<ImageView>(R.id.imgLarge)
        grid.adapter = ImageAdapter(this)
        grid.onItemClickListener = AdapterView.OnItemClickListener { _, _, position, _ ->
            Toast.makeText(baseContext, "Selected Species " + (position + 1), Toast.LENGTH_SHORT).show()
            pic.setImageResource(animals[position])
        }
    }

    inner class ImageAdapter internal constructor(private val context: Context) : BaseAdapter() {

        override fun getCount(): Int {
            return animals.size
        }

        override fun getItem(position: Int): Any? {
            return null
        }

        override fun getItemId(position: Int): Long {
            return 0
        }

        override fun getView(position: Int, convertView: View, parent: ViewGroup): View {
            pic = ImageView(context)
            pic.setImageResource(animals[position])
            pic.scaleType = ImageView.ScaleType.FIT_XY
            //pic.layoutParams = GridView.LayoutParams(330, 300)
            return pic
        }
    }
}

